src/test/java/lambda/part1/exercise/Exercise3.java

```diff
@@ -0,0 +1,64 @@
+package lambda.part1.exercise;
+
+import com.google.common.collect.FluentIterable;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise3 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, (o1, o2) -> Integer.compare(o1.getAge(), o2.getAge()));
+
+        assertArrayEquals(new Person[]{
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, (o1, o2) -> {
+            return o1.getLastName().equalsIgnoreCase(o2.getLastName()) ? o1.getFirstName().compareToIgnoreCase(o2.getFirstName())
+                    : o1.getLastName().compareToIgnoreCase(o2.getLastName());
+        });
+
+        assertArrayEquals(new Person[]{
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45),
+                new Person("Николай", "Зимов", 30),
+                new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicateLambda() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Person person = FluentIterable.from(persons).firstMatch(p -> new Integer(30).equals(p.getAge())).get();
+
```

Comment:
As i understood it's supposed that you should create Predicate using lambda expression, and then use it in firstMath() method.

Reply:
it will have not so many differencies, just need to extract Guava Predicate